### PR TITLE
in_forward: Check unix socket availability 

### PR DESCRIPTION
**Pull request by cosmo0920**
Windows cannot handle unix socket.
We should disable it on Windows.

Signed-off-by: Hiroshi Hatake <hatake@clear-code.com>

### CODE REVIEW

**Comment by edsiper** [comment]
I am wondering if is better to disable in_forward in Windows.

**Comment by cosmo0920** [comment]
Umm, I'm not sure we should disable it on Windows.  :confounded:

**Comment by edsiper** [comment]
what about to patch /CMakeLists.txt and add a discovery routine for unix socket and set a "FLB_HAVE_UNIX_SOCKET" ? on that way the plugin can just drop that unix socket support

File: CMakeLists.txt [hunk 1]
```diff
@@ -377,6 +377,20 @@ if(FLB_HAVE_CLOCK_GET_TIME)
   FLB_DEFINITION(FLB_HAVE_CLOCK_GET_TIME)
 endif()
 
+# unix socket support
+check_c_source_compiles("
+  #include <sys/types.h>
+  #include <sys/socket.h>
+  int main() {
+      int sock;
+      sock = socket(AF_INET, SOCK_STREAM, 0);
```
Comment (edsiper) [inline CMakeLists.txt]: it should be AF_UNIX, otherwise its creating a normal TCP socket.

Comment (cosmo0920) [reply CMakeLists.txt]: Oops. I've fixed it.

**Comment by edsiper** [comment]
in order to merge please do 2 commits:

1. subject => build: detect if unix socket is available FLB_HAVE_UNIX_SOCKET
2. subject => in_forward: disable unix socket support if not available in the system

**Comment by cosmo0920** [comment]
I updated this PR to follow your suggestion.

**Comment by edsiper** [comment]
thanks

**Pull request merged**